dhis-2/dhis-services/dhis-service-tracker/src/main/java/org/hisp/dhis/tracker/imports/validation/validator/event/SecurityOwnershipValidator.java

```diff
@@ -231,9 +233,9 @@ private String getTeiUidFromEvent(
           .findEnrollmentByUid(event.getEnrollment())
           .map(org.hisp.dhis.tracker.imports.domain.Enrollment::getTrackedEntity)
           .orElse(null);
-    } else {
-      return enrollment.getTrackedEntity().getUid();
     }
+
+    return enrollment.getTrackedEntity() != null ? enrollment.getTrackedEntity().getUid() : null;
```

Comment:
well, in case we removed the validator, we would probably do a different setup. But you are right as we are not getting there so we don't need this change